sklearn/preprocessing/_encoders.py

```diff
@@ -281,6 +284,8 @@ def _identify_infrequent(self, category_count, n_samples, col_idx):
             If there are infrequent categories, indices of infrequent
             categories. Otherwise None.
         """
+        # TODO ohe_sw: We would have to change this...
+        # But it really makes sense... :(
         if isinstance(self.min_frequency, numbers.Integral):
             infrequent_mask = category_count < self.min_frequency
         elif isinstance(self.min_frequency, numbers.Real):
```

Comment:
I think I'll change the structure to something like: if `sample_weight is None` then behave the same, else then interpret as the "int version", ie, `category_count < self.min_frequency` (in this case, `category_count` is the sum of `sample_weight` of that category).